docs/DeveloperGuide.md

```diff
@@ -1,34 +1,978 @@
+
+The UI component is made up of main 2 classes:
+* `TextUi`: Responsible for the majority display of Quotesify’s messages. 
+* `UiMessage`: Holds all messages required for TextUI to print to the program console.
+
+In essence, the UI is responsible for the majority display of all successful command executions, error messages
+, as well as user interaction by prompting for the next command.
+
+### 3.3 Logic Component
+![Class Diagram for Logic Component](images/ClassDiagram_Logic.png)
+
+The Logic component is made up of 2 sub-components, namely the `Parser` and `Command`. Below describes the sequence
+flow from the time a user input is read till command execution ends.
+
+1. User input is fetched from the UI and passed into the Parser for parsing.
+2. A Command object is returned and subsequently executed in the Main class.
+3. The command execution outcome may affect Quotesify’s model objects. (e.g. adding a book)
+4. Command instructs the UI component to print out relevant output messages depending on command type.
+Also, Command may invoke saving of data via Storage at a given point in time.
+5. Finally, Command will then inspect the exit status after command execution.
+6. Control is handed back over to the UI.
+
+* `XCommand` class represents the command type such as `AddCommand`, `FindCommand`, `DeleteCommand`, etc.
+* `XYCommand` class represents the command to execute from a specific object. Such as `AddBookCommand`
+, `FindBookCommand`, `DeleteQuoteCommand`, etc.
+
+### 3.4 Model Component
+![Class Diagram for Model Component](images/ClassDiagram_Model.png)
+
+The model component consists of several classes that make up the main features of Quotesify.
+Each object holds in-application data unique to each feature and is stored in a list of their own.
+
+### 3.5 Storage Component
+The storage component consists of a single `Storage` class. It is responsible for saving user data as instructed
+by the command component as well as to detect and load data on program launch.
+
+On program launch:
+1. `Storage` is initialised and checks for the existence of save data.
+2. If save data exists, `Storage` will read the save data in JSON format and parses them back into their
+model objects (e.g. Book).
+3. If save data does not exist, `Storage` will create an empty save file in the specified directory.
+
+On Command execution:
+1. `Storage` parses all model objects in JSON format and writes into the save file.
+
+## 4.0 Implementation
+*`Note: All UML diagrams in this guide are stored in the images/ directory.`*
+
+### 4.1 Feature: Book Management
+Given below is the class diagram for classes related to Book Management in Quotesify:
+
+![Class Diagram for Book Management](images/ClassDiagram_Book.png)
+
+#### 4.1.1 Add Books
+The sequence diagram below demonstrates the command execution process when adding a book to the booklist.
+
+![Sequence Diagram for Add Books](images/SeqDiagram_AddBook.png)
```

Comment:
![Screenshot 2020-10-28 at 11 40 21 AM](https://user-images.githubusercontent.com/13443360/97388116-5f113100-1912-11eb-914a-133aeab3dbb7.png)

Hello, nice diagram! Should there be the name of the method called on BookList by ListManager?